Adafruit_NeoPixel.cpp

```diff
@@ -2163,8 +2167,16 @@ void Adafruit_NeoPixel::show(void) {
   @param   p  Arduino pin number (-1 = no pin).
 */
 void Adafruit_NeoPixel::setPin(uint16_t p) {
-  if(begun && (pin >= 0)) pinMode(pin, INPUT);
+  if(begun && (pin >= 0)) {
+#if defined(AM_PART_APOLLO3)
+    apollo3UnsetPin(pin);
+#endif
+    pinMode(pin, INPUT);
+  }
     pin = p;
+#if defined(AM_PART_APOLLO3)
+    apollo3SetPin(pin);

```

Comment:
After updating the ```pin``` variable either ```pin``` or ```p``` can be used to refer to the new pin number. The usage in the standard ```pinMode``` call uses ```p``` while the specialized call for Apollo3 uses ```pin```. This may be more confusing/harder to maintain in the future. Because ```p``` is already in use my rec. would be to use ```p``` in the specialized call as well

Reply:
Agreed, although now I'm modifying the value of `p` to the Apollo3 mapped pad, which might also be slightly confusing.